.github/workflows/auto-recode.yml

```diff
@@ -0,0 +1,60 @@
+name: pr-number-assign
+on: 
+  pull_request:
+permissions:
+  contents: write
+  
+jobs:
+  pr-number-assign:
+    runs-on: ubuntu-latest 
+    
+    env:
+      FEAT: default
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.head_ref }}
+    
+    - name: do work
+      run: |
+        script=$(cat <<EOF
+        import os
+
+        pr_number = "${{github.event.pull_request.number}}"
+        path = "/home/runner/work/backend.ai/backend.ai/changes"
+        t_types = ["breaking","feature","enhance","deprecation","fix","doc","deps","misc"]
+        target_file = ""
+
+        files = os.listdir(path)
+        print(str(files) + " are in changes floder.")
+
+        for file in files :
+            if str(file[0:file.find(".")]) == pr_number :
+                print("Change log file already exist for this PR.")
+                exit(0)
+
+        for t_type in t_types :
+            if t_type in files : 
```

Comment:
It doesn't seem to handle the case of writing anything but a pr number, like `doc.md`.

Reply:
I wanted to make my workflow as 'optional' thing for developers for the reason that they can check pr number manually as they did. I don't want to bothering them to use this workflow. Under this opinion, I made this workflow only change file name 'feat' to 'pr_number.feat.md'.

As you can see, there is not error case in my workflow because there is timeline-check workflow that check if there is pr number in file name. So 'doc.md' case will be treated as last case and it will print "There is not change log file for this PR in changes folder." with no error in my workflow and make error in timeline check workflow. This means my workflow don't need to consider such a case like 'doc.md'.

Reply:
It should support both ".<category>" and ".<category>.md" filename patterns.

Reply:
I think it may be better to use regex, such as:
```python
if match := re.search(r"\." + re.escape(t_type) + r"(\.md)?", filename):
    ...
```

Reply:
Also I'd recommend to use `pathlib` instead of `os` and `os.path` APIs, to get used with modern Python path manipulation APIs.